alphagov/paas-cf-apps-statsd
# [#153628648] Add METRIC_WHITELIST option

# What

We have added a new option to allow filtering which metrics are emitted by the exporter.

You can set the option via `--metric-whitelist` cli arg or `METRIC_WHITELIST` env var. which accepts a csv of prefixes of metric names.

# How to review

* See the README
* Check the code is sane
* Check the tests are good
* Try out in debug mode against your environment:

```
go run main.go --api-endpoint "https://api.${DEPLOY_ENV}.dev.cloudpipeline.digital" --skip-ssl-validation --username admin --password ${UAA_ADMIN_PASSWORD} --debug --metric-whitelist "cpu,requests"
```

# Who can review

Not @chrisfarms nor @paroxp
